docs/topics/custom-images.md

```diff
@@ -0,0 +1,103 @@
+# Custom images
+
+This document will help you get a CAPZ Kubernetes cluster up and running with your custom image.
+
+## Reference images
+
+An *image* defines the operating system and Kubernetes components that will populate the disk of each node in your cluster.
+
+By default, images offered by "capi" in the [Azure Marketplace][azure-marketplace] are used. You can list these *reference images* with this command:
+
+```bash
+az vm image list --publisher cncf-upstream --offer capi --all -o table
+```
+
+But for more control over your nodes, you should build, publish, and use a *custom image*.
+
+## Building a custom image
+
+Cluster API uses the Kubernetes [Image Builder][image-builder] tools. You can use the [Azure image][image-builder-azure] from that project as a starting point for your custom image.
+
+[The Image Builder Book][capi-images] explains how to customize and build the images defined in that repository, with instructions for [Azure CAPI Images][azure-capi-images] in particular.
+
+### Operating system expectations
+
+TODO!
+
+### Required components
+
+TODO!
```

Comment:
A lot of these requirements will depend on the bootstrap provider, eg. kubeadm has these OS requirements https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin and kubelet, kubeadm and kubectl binaries https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl

Reply:
Let's also add a section about k8s version expectations (mainly that the image k8s version needs to match the k8s version of the Machine / KubeadmControlPlane) and the implication on k8s version upgrades - to upgrade your Machines you need to create a new custom image with the k8s version you want to upgrade to and update that image on your Machines / KCPs  alongside `Version`.